src/ConfigAggregator.php

```diff
@@ -60,19 +60,21 @@ class ConfigAggregator
      * @param array $postProcessors Array of processors. These may be callables, or
      *     string values representing classes that act as processors. If the
      *     latter, they must be instantiable without constructor arguments.
+     * @param int $mode File mode to set on cached config
      */
     public function __construct(
         array $providers = [],
         $cachedConfigFile = null,
-        array $postProcessors = []
+        array $postProcessors = [],
+        $mode = 0644
```

Comment:
This is octal 644, which is decimal 420.

Reply:
There are quite a few setups where server is run as different user, sharing group with cli user. I believe this stricter mode might cause breakage for some flows.

To be safe I would prefer default to be null that will not try to set any mode, keeping current behavior where system umask controls the mode.